### PR TITLE
Added Solution in Python for 0023 - Merge K Sorted Lists.

### PR DESCRIPTION
**Pull request by radojicic23**
## Change Summary

The purpose of this PR is to add coding solution in Python for the LeetCode [Merge K Sorted Lists](https://leetcode.com/problems/merge-k-sorted-lists/) problem.

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Solutions

- [x] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [x] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [x] I've confirmed that comments are put above each line rather than writing on the same line.
- [x] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

### CODE REVIEW

**Comment by radojicic23** [comment]
Can you check this @wingkwong ? Thank you.

File: solutions/0000-0099/0023-merge-k-sorted-lists-hard.md [hunk 1]
```diff
@@ -166,4 +166,58 @@ class Solution {
...
+
+```python
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
+            # Edgecases: if length is 0 and lists is None
```
Comment (wingkwong) [inline solutions/0000-0099/0023-merge-k-sorted-lists-hard.md:180]: indentation

File: solutions/0000-0099/0023-merge-k-sorted-lists-hard.md [hunk 2]
```diff
@@ -166,4 +166,58 @@ class Solution {
...
+        if not lists or len(lists) == 0:
+            return None
+        # Taking pairs of LL and merging them each time and
```
Comment (wingkwong) [inline solutions/0000-0099/0023-merge-k-sorted-lists-hard.md:183]: Taking -> take

**Review by wingkwong — CHANGES_REQUESTED** [review]
for all comments, please start with lower case

**Pull request merged**